state-sync/storage-service/server/src/lib.rs

```diff
@@ -364,18 +392,24 @@ impl StorageReaderInterface for StorageReader {
         &self,
         _version: u64,
         _start_account_index: u64,
-        _expected_num_account_states: u64,
+        _end_account_index: u64,
     ) -> Result<AccountStatesChunkWithProof, Error> {
         // TODO(joshlind): implement this once DbReaderWriter supports these calls.
         Err(Error::UnexpectedErrorEncountered(
             "Unimplemented! This API call needs to be implemented!".into(),
         ))
     }
+}
 
-    fn get_number_of_accounts(&self, _version: u64) -> Result<u64, Error> {
-        // TODO(joshlind): implement this once DbReaderWriter supports these calls.
-        Err(Error::UnexpectedErrorEncountered(
-            "Unimplemented! This API call needs to be implemented!".into(),
-        ))
-    }
+/// Calculate `(start..=end).len()`. Returns an error if `end < start` or
+/// `end == u64::MAX`.
+fn range_len(start: u64, end: u64) -> Result<u64, Error> {
+    // len = end - start + 1
+    let len = end.checked_sub(start).ok_or_else(|| {
+        Error::InvalidRequest(format!("end ({}) must be >= start ({})", end, start))
+    })?;
+    let len = len
+        .checked_add(1)
+        .ok_or_else(|| Error::InvalidRequest(format!("end ({}) must not be u64::MAX", end)))?;
```

Comment:
This is a function Philip recently added. I've simply moved it in this PR, but:
- `u64::MAX - 1` is a valid version request, so I don't see why we'd want to limit/prevent it?
- If our versions ever actually got that high, I suspect our code would fail in a million places before it would here, so I'm inclined to leave this as it is 😄